includes/Jobs/DataDumpGenerateJob.php

```diff
@@ -5,176 +5,360 @@
-		}
+		$this->arguments = $params['arguments'];
+		$this->fileName = $params['fileName'];
+		$this->type = $params['type'];
 
-		$logEntry->setParameters( [ '4::filename' => $fileName ] );
-		$logEntry->publish( $logEntry->insert() );
+		$this->config = $configFactory->makeConfig( 'DataDump' );
+		$this->connectionProvider = $connectionProvider;
 	}
 
-	public function run() {
-		$dataDumpConfig = $this->config->get( 'DataDump' );
-		$dataDumpLimits = $this->config->get( 'DataDumpLimits' );
+	public function run(): bool {
+		$dataDumpConfig = $this->config->get( ConfigNames::DataDump );
+		$dataDumpLimits = $this->config->get( ConfigNames::Limits );
 		$dbName = $this->config->get( MainConfigNames::DBname );
+		$dbw = $this->connectionProvider->getPrimaryDatabase();
 
-		$dbw = MediaWikiServices::getInstance()
-			->getDBLoadBalancer()
-			->getMaintenanceConnectionRef( DB_PRIMARY );
+		$fileName = $this->fileName;
+		$type = $this->type;
 
-		$fileName = $this->params['fileName'];
-		$type = $this->params['type'];
+		$this->setStatus(
+			status: 'in-progress',
+			dbw: $dbw,
+			directoryBackend: '',
+			fileName: $fileName,
+			fname: __METHOD__,
+			comment: '',
+			fileSize: 0
+		);
 
-		$this->setStatus( 'in-progress', $dbw, '', $fileName, __METHOD__ );
-
-		$options = [];
-		foreach ( $dataDumpConfig[$type]['generate']['options'] as $option ) {
-			$options[] = preg_replace( '/\$\{filename\}/im', $fileName, $option );
-		}
+		$options = array_map(
+			fn ( string $opt ): string => preg_replace(
+				'/\$\{filename\}/im', $fileName, $opt
+			),
+			$dataDumpConfig[$type]['generate']['options'] ?? []
+		);
 
 		$dataDumpConfig[$type]['generate']['options'] = $options;
 
 		$backend = DataDump::getBackend();
 		$directoryBackend = $backend->getContainerStoragePath( 'dumps-backup' );
+
 		if ( !$backend->directoryExists( [ 'dir' => $directoryBackend ] ) ) {
 			$backend->prepare( [ 'dir' => $directoryBackend ] );
 		}
 
-		if ( ( $dataDumpConfig[$type]['generate']['type'] ?? '' ) === 'mwscript' ) {
-			$generate = array_merge(
-				$dataDumpConfig[$type]['generate']['options'],
-				$this->params['arguments'] ?? [],
-				[ '--wiki', $dbName ]
+		$result = $this->executeCommand(
+			config: $dataDumpConfig,
+			limits: $dataDumpLimits,
+			type: $type,
+			dbName: $dbName,
+			fileName: $fileName
+		);
+
+		if ( $result === 0 ) {
+			return $this->handleSuccess(
+				config: $dataDumpConfig,
+				type: $type,
+				fileName: $fileName,
+				directoryBackend: $directoryBackend,
+				dbw: $dbw
 			);
+		}
 
-			$result = Shell::makeScriptCommand(
-				$dataDumpConfig[$type]['generate']['script'] ?? '',
-				$generate
-			)->limits( $dataDumpLimits )
-				->disableSandbox()
-				->includeStderr()
-				->execute()
-				->getExitCode();
-		} else {
+		return $this->setStatus(
+			status: 'failed',
+			dbw: $dbw,
+			directoryBackend: $directoryBackend,
+			fileName: $fileName,
+			fname: __METHOD__,
+			comment: 'Something went wrong',
```

Comment:
What about a more explanatory message with the status code, such as something like `"Something went wrong: Command exited with {$result}"`